app/proxy_info.py

```diff
@@ -99,7 +107,28 @@ def parse_proxy_info(proxy_info_file: str = cfg.PROXY_INFO_FILE) -> ProxyInfo:
     # this option will be set to the default value
     _proxy_info_dict: Dict[str, Any] = dict()
     for _field in fields(ProxyInfo):
-        _proxy_info_dict[_field.name] = _loaded.get(_field.name, _field.default)
+        _option = _loaded.get(_field.name)
+        if not isinstance(_option, _field.type):
+            if _option is not None:
+                logger.warning(
+                    f"{_field.name} contains invalid value={_option}, ignored and set to default={_field.default}"
+                )
+            _proxy_info_dict[_field.name] = _field.default
+            continue
+
+        _proxy_info_dict[_field.name] = _option
+
+    # maintain compatiblity with old proxy_info format
+    for old, new in ProxyInfo._compatibility.items():
+        if old in _loaded:
+            warnings.warn(
+                "'enable_ota_proxy' is replaced by 'enable_local_ota_proxy', "
+                "and the support for 'enable_ota_proxy' option might be dropped in the future "
+                "please use 'enable_local_ota_proxy' instead.",
```

Comment:
This should be fine.
```suggestion
            warnings.warn(
                f"'{old}' is replaced by '{new}', "
                f"and the support for '{old}' option might be dropped in the future "
                f"please use '{new}' instead.",
```

Reply:
Ah yes! I make a generic method but hardcode the value, what a miss!